### ISSUE TITLE
When breeze.js is loaded via CommonJS, Q should be included as a `package.json` dependency and loaded with `require('q');`

### ISSUE THREAD

**Issue by dimlock** [Breeze/breeze.js#119]
Assuming the `require("breeze")` [issue](https://github.com/Breeze/breeze.js/pull/118) is fixed, it would be better if Q was included as a package.json dependency and loaded with a require('q'); within breeze.js when loaded via CommonJS. Expecting Q to be available globally (i.e. window.Q) goes against a fundamental benefit of CommonJS and modular JS code.

**Comment by steveschmitt** [comment]
True, except that breeze doesn't necessarily need Q, it just needs a compatible promise implementation.  For example, [Angular's $q should be used in Angular applications](http://breeze.github.io/doc-js/breeze-angular.html#breeze-promises).

**Comment by phunksta** [comment]
Very much bumping into this right now.
In attempting to implement something similar to the example found here: [breeze-odata4]https://github.com/tschettler/breeze-odata4-samples/blob/master/angular-breeze-odata4/src/app/app.component.ts

at line 57 I find code like
```
    this.manager.fetchMetadata().then(() => {
      this.entityTypes = <EntityType[]>this.manager.metadataStore.getEntityTypes()
        .filter((et: any) => (et instanceof EntityType) && !!et.defaultResourceName);
    });
```

In the absence of using angular, rather browserify and plain old javascript, the fetchMetadata() function doesn't return the promise when the Q library isn't present. The code never gets to the function in .then() and although the metadata was fetched from the service, nothing else seems to happen.
As Q seems to be deprecated, at least in npm, is there some way to shim this so it works?
